x-pack/plugins/upgrade_assistant/__jest__/client_integration/elasticsearch.test.ts

```diff
@@ -0,0 +1,777 @@
+          details:
+            'model snapshot [%s] for job [%s] supports minimum version [%s] and needs to be at least [%s]',
+          url: 'doc_url',
+          correctiveAction: {
+            type: 'mlSnapshot',
+            snapshotId,
+            jobId,
+          },
+        })
+      );
+
+      const indexSettingsDeprecations: EnrichedDeprecationInfo[] = Array.from(
+        {
+          length: 20,
+        },
+        () => ({
+          isCritical: false,
+          resolveDuringUpgrade: false,
+          type: 'index_settings',
+          message: indexSettingDeprecations.translog.deprecationMessage,
+          details: 'deprecation details',
+          url: 'doc_url',
+          index: 'my_index',
+          correctiveAction: {
+            type: 'indexSetting',
+            deprecatedSettings: indexSettingDeprecations.translog.settings,
+          },
+        })
+      );
+
+      const reindexDeprecations: EnrichedDeprecationInfo[] = Array.from(
+        {
+          length: 20,
+        },
+        () => ({
+          isCritical: true,
+          resolveDuringUpgrade: false,
+          type: 'index_settings',
+          message: 'Index created before 7.0',
+          details: 'deprecation details',
+          url: 'doc_url',
+          index: 'reindex_index',
+          correctiveAction: {
+            type: 'reindex',
+          },
+        })
+      );
+
+      const defaultDeprecations: EnrichedDeprecationInfo[] = Array.from(
+        {
+          length: 20,
+        },
+        () => ({
+          isCritical: false,
+          resolveDuringUpgrade: false,
+          type: 'index_settings',
+          message: 'multi-fields within multi-fields',
+          details: 'deprecation details',
+          url: 'doc_url',
+          index: 'nested_multi-fields',
+        })
+      );
+
+      const deprecations: EnrichedDeprecationInfo[] = [
+        ...defaultDeprecations,
+        ...reindexDeprecations,
+        ...indexSettingsDeprecations,
+        ...mlDeprecations,
+      ];
+
+      const upgradeStatusWithManyDeprecations: ESUpgradeStatus = {
+        totalCriticalDeprecations: 40,
+        deprecations,
+      };
+
+      beforeEach(async () => {
+        httpRequestsMockHelpers.setLoadEsDeprecationsResponse(upgradeStatusWithManyDeprecations);
+        httpRequestsMockHelpers.setUpgradeMlSnapshotStatusResponse({
+          nodeId: 'my_node',
+          snapshotId,
+          jobId,
+          status: 'idle',
+        });
+
+        await act(async () => {
+          testBed = await setupElasticsearchPage({ isReadOnlyMode: false });
+        });
+
+        testBed.component.update();
+      });
+
+      it('shows the correct number of pages and deprecations per page', async () => {
+        const { find, actions } = testBed;
+
+        expect(find('esDeprecationsPagination').find('.euiPagination__item').length).toEqual(
+          Math.round(deprecations.length / 50) // Default rows per page is 50
```

Comment:
nit: seems kibana deprecations table defaults to 25 (defined in the constants file) but es deprecations table to 50. maybe we should make them default to the same number? and import this `DEPRECATIONS_PER_PAGE` value in here 🤔

Reply:
Good catch! After an initial design review with CJ and Dmitry, we decided to change the default number of pages to `50`. I'll update the Kibana deprecations to align as part of https://github.com/elastic/kibana/pull/109150.